MapboxNavigation/MapboxSpeechSynthesizer.swift

```diff
@@ -25,9 +25,12 @@ open class MapboxSpeechSynthesizer: NSObject, SpeechSynthesizing {
     public var isSpeaking: Bool {
         return audioPlayer?.isPlaying ?? false
     }
-    public var locale: Locale = Locale.autoupdatingCurrent
+    public var locale: Locale? = Locale.autoupdatingCurrent
```

Comment:
What does it mean if the developer sets `locale` to `nil`, given that the default value is already `autoupdatingCurrent`? If we should retain this edge case instead of falling back to `autoupdatingCurrent`, then let’s document it.

Reply:
For vocalizing an instruction a property`locale` or a 'locale' argument to corresponding `speak` method can be used.
I was thinking about `SpeechSynthesizing` usage scenario as to provide user a flexible ways to provide `Locale`s: 
* it can be set by default for all instructions (property is set, method argument is nil)
* it can be overridden for a specific instruction (property is set, method argument set)
* it can be required to manually specify `Locale` for each instruction (property is nil, method argument is set)

The latter case requires property to by `optional`
For `MapboxSpeechSynthesizer` I decided to stick to the first variant, so I've added a default value.